### PR TITLE
add testcases for create , delete and rollback to snapshot

### PR DESCRIPTION
**Pull request by Dinaamagdy**

### CODE REVIEW

File: tests/integration_tests/testsuite/ovc_tests/a_basic/vms_tests.py [hunk 1]
```diff
@@ -302,6 +302,83 @@ def test004_clone_vm(self):
...
+        """
+        self.log('%s STARTED' % self._testID)
+        
+        self.log("Create snapshots[sn1],[sn2] and [sn3] of [vm1], should succeed.")
+        temp_actions = {'node': {'actions': ['snapshot'], 'service': self.vm1}}
+        for _ in range(3):
+            res = self.create_vm(accounts=self.accounts, cloudspaces=self.cloudspaces,
+                                 vms=self.vms, temp_actions=temp_actions)
+            self.wait_for_service_action_status(self.vm1, res[self.vm1]['snapshot'])
+            time.sleep(2)
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/ovc_tests/a_basic/vms_tests.py]: why sleep is needed inside the for?

File: tests/integration_tests/testsuite/ovc_tests/a_basic/vms_tests.py [hunk 2]
```diff
@@ -302,6 +302,83 @@ def test004_clone_vm(self):
...
+        self.log("Rollback to [sn2], should succeed.")
+        temp_actions = {'node': {'actions': ['snapshot_rollback'], 'service': self.vm1,
+                                 'args': {'snapshot_epoch': snapshots_before_rollback[1]["epoch"]}}}
+        res = self.create_vm(accounts=self.accounts, cloudspaces=self.cloudspaces,
+                             vms=self.vms, temp_actions=temp_actions)       
+
+        self.log("Check that [sn3] has been deleted .")
+        snapshots_after_rollback = self.get_snapshots_list(self.cs1, self.vm1)
+        self.assertNotIn(snapshots_before_rollback[2]["name"], [sn["name"] for sn in snapshots_after_rollback])
+
```
Comment (john-kheir) [inline tests/integration_tests/testsuite/ovc_tests/a_basic/vms_tests.py]: also check on the length of snapshots

File: tests/integration_tests/testsuite/ovc_tests/a_basic/vms_tests.py (hunk 1 above)
Comment (Dinaamagdy) [reply tests/integration_tests/testsuite/ovc_tests/a_basic/vms_tests.py]: to add an amount of time between snapshots

**Comment by Dinaamagdy** [comment]
----------------------------------------------------------------------
Ran 20 tests in 364.557s

OK (SKIP=17)

**Pull request merged**